datasource/src/main/scala/quasar/plugin/sqlserver/datasource/SQLServerDatasource.scala

```diff
@@ -69,4 +77,50 @@ private[datasource] object SQLServerDatasource {
       SQLServerDatasourceModule.kind,
       NonEmptyList.one(Loader.Batch(loader)))
   }
+
+  private def offsetFragment(offset: Offset): Either[String, Fragment] =
+    for {
+      ioffset <- internalize(offset)
+      cfr <- columnFragment(ioffset)
+      result <- makeOffset(cfr, ioffset.value)
+    } yield result
+
+  private def internalize(offset: Offset): Either[String, Offset.Internal] = offset match {
+    case _: Offset.External =>
+      Left("SQL Server datasource supports only internal offsets")
+    case i: Offset.Internal =>
+      i.asRight[String]
+  }
+
+  private def columnFragment(offset: Offset.Internal): Either[String, Fragment] =
+    offset.path match {
+      case NonEmptyList(DataPathSegment.Field(s), List()) =>
+        Fragment.const(SQLServerHygiene.hygienicIdent(Ident(s)).forSql).asRight[String]
+      case _ =>
+        Left("Incorrect offset path")
+    }
+
+  private def makeOffset(colFr: Fragment, key: ∃[InternalKey.Actual]): Either[String, Fragment] = {
+    val actual: InternalKey[Id, _] = key.value
+
+    val actualFragment = actual match {
+      case InternalKey.RealKey(k) =>
+        Right(Fragment.const(k.toString))
+      case InternalKey.StringKey(s) =>
+        Right(fr0"'" ++ Fragment.const0(s.replace("'", "''")) ++ fr0"'")
+      case InternalKey.DateTimeKey(d) =>
+        Right(fr0"'" ++ Fragment.const0(d.toString) ++ fr0"'")
+      case InternalKey.LocalDateTimeKey(d) =>
+        Right(fr0"'" ++ Fragment.const0(d.toString) ++ fr0"'")
+      case InternalKey.LocalDateKey(d) =>
+        Right(fr0"'" ++ Fragment.const0(d.toString) ++ fr0"'")
```

Comment:
These `.toString`s are ok as well?

Reply:
Yep, the same as in the other db, we're OK with `YYYY-MM-DD`